### DISCUSSION TITLE
【要望】コメントした記事を検索できるようにして欲しい

### DISCUSSION THREAD

**Discussion by nak435** [increments/qiita-discussions#734] [要望・アイデア・不具合]
`answer_by:ユーザ`で回答した質問を検索できるように、
`comment_by:ユーザ`でコメントした記事を検索できるようにして欲しい。
（検索キーワードはお任せします）

自分が過去にコメントした記事を探したいときがあり、今は`tag`や`title`で探しているが、「コメントした記事を検索できる」と大変便利です。
マイページから「コメントした記事」の一覧は見れますが、並べ替えや検索ができないため、過去の記事を探すのが大変。

ご検討のほど、よろしくお願いします。

**Comment by tanaka-asami415** [comment]
@nak435 
リクエストありがとうございます！
直近の開発優先度などから着手は先になりますが、我々も検索できるようにしたいと思っておりますので、気長にお待ちいただけると幸いです🙇

**Reply by nak435** [reply]
先ほど検索オプションに`answer_by`が追加されたことを知りました。
正式アナウンスされてましたか？
（他に追加された検索オプションがあれば教えてください）

**Reply by tanaka-asami415** [reply]
@nak435 
`answer_by:ユーザ` は質問を検索できるようにした際に検索可能となり、以前より検索可能なオプションです。
（こちらのDiscussionsをいただいてから、新たに追加された検索オプションはまだございません🙇）

**Reply by nak435** [reply]
そうでした。失礼しました。

**Reply by tanaka-asami415** [reply]
@nak435 
お待たせしております🙇
リクエストの実装に向けて仕様を検討している段階なのですが、現在想定している仕様でご要望を叶えられる状態なのか、確認のため質問させてください🙏

過去にコメントした記事を探したいときがあるということでしたが、「コメントした記事を検索したい」時というのは、具体的にはどのような時でしょうか？

例えば、
- 自分がどんなコメントをしたのか確認したいために、コメント元の記事を検索したい（コメントを探す事が目的）
- 自分がコメントした記事をもう一度読みたい（記事を探す事が目的）
- 他のユーザーがコメントした記事も見たい

など…お聞かせいただけると助かります！

**Reply by nak435** [reply]
・自分がどんなコメントをしたのか確認したいために、コメント元の記事を検索したい（コメントを探す事が目的）
・自分がコメントした記事をもう一度読みたい（記事を探す事が目的）
両方ともあると思います（思ったことがあります）。
よろしくお願いします。

**Comment by tanaka-asami415** [comment]
nak435さん以外にも「コメント関連の検索をしたい時がある」という方がいらっしゃいましたら、ぜひ教えていただけると幸いです！🙇

**Reply by shiracamus** [reply]
私も自分のコメントを検索できると嬉しいです。
「参照渡し」に関して沢山コメントしてきたのを検索したいと思ったことが何度もあります。

**Comment by tanaka-asami415** [comment]
@nak435 @shiracamus

ありがとうございます！ご返信が遅くなり申し訳ありません🙇

検討していた仕様でご要望に対しては問題なさそうです！

しかしながら、仮実装してみたところ、かなり開発工数がかかる事が判明いたしました…😢
直近の優先度からは実装はかなり先になってしまいそうなため、他にいい方法がないかも検討したいと思います。

実装までお時間がかかると思いますが、気長にお待ちいただけると幸いです。
